packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -538,7 +538,7 @@ const Autocomplete = React.forwardRef(function Autocomplete(inProps, ref) {
             ref: setAnchorEl,
             className: classes.inputRoot,
             startAdornment,
-            endAdornment: (
+            endAdornment: (hasClearIcon || hasPopupIcon) && (
```

Comment:
Let's not do this because if it is evaluated to `false`, the `endAdornment` will have `false` value (`endAdornment: false`). Instead use short-circuiting with spread operator:
```suggestion
            ...(hasClearIcon ||
              (hasPopupIcon && {
              endAdornment: (
```